enterprise/frontend/src/embedding-sdk/cli/snippets/analytics-provider-snippet.ts

```diff
@@ -23,8 +37,19 @@ import {createContext, useCallback, useEffect, useState} from 'react'
 
 export const BASE_SSO_API = 'http://localhost:4477'
 
-// Used for the example theme switcher and user switcher components
-export const AnalyticsContext = createContext({})
+/**
+ * @typedef {Object} AnalyticsContextType
+ * @property {'light'|'dark'} themeKey - The current theme key.
+ * @property {(themeKey: 'light'|'dark') => void} setThemeKey - Function to update the theme.
+ * @property {string|null} [email] - Email of the user.
+ * @property {(user: string) => void} [switchUser] - Function to switch users.
+ * @property {string|null} [authError] - Optional auth error message.
+ */
+
+/**
+ * @type {React.Context<AnalyticsContextType>}
+ */
+export const AnalyticsContext = createContext(/** @type {AnalyticsContextType} */ ({}));
```

Comment:
I think the inner `{AnalyticsContextType}` can be inferred so we don't need it

Reply:
The same as above

Reply:
can we get rid of the [outer React.Context type](https://github.com/metabase/metabase/pull/54972#discussion_r1993041447) instead?